cpp/include/nvtx3/nvtx3.hpp

```diff
@@ -2086,8 +2161,12 @@ class domain_process_range {
   domain_process_range& operator=(domain_process_range const&) = delete;
 
  private:
-  std::unique_ptr<range_handle> handle_;  ///< Range handle used to correlate
-                                          ///< the start/end of the range
+  struct end_range_handle {
+    using pointer = range_handle;  /// Override the pointer type of the unique_ptr
+    void operator()(range_handle h) const noexcept { end_range(h); }
```

Comment:
I think we should stick with using domains for range ends.  So, I guess, make this struct a template so you can specify D, call end_range<D> in operator(), and then use end_range_handle<D> for the unique_ptr's deleter.

Reply:
This is handled in https://github.com/NVIDIA/NVTX/pull/37